Pod/Classes/KolodaView/SwipeResultDirection.swift

```diff
@@ -0,0 +1,183 @@
+        let h = VerticalPosition.Bottom.rawValue - VerticalPosition.Top.rawValue
+        return CGRect(x: HorizontalPosition.Left.rawValue, y: VerticalPosition.Top.rawValue, width: w, height: h)
+    }
+}
+
+
+private enum VerticalPosition:CGFloat {
+    case Top = -1
+    case Middle = 0
+    case Bottom = 1
+}
+
+private enum HorizontalPosition:CGFloat {
+    case Left = -1
+    case Middle = 0
+    case Right = 1
+}
+
+
+private struct Direction {
+    let horizontalPosition:HorizontalPosition
+    let verticalPosition:VerticalPosition
+    
+    var point:CGPoint {
+        return CGPoint(x:horizontalPosition.rawValue, y: verticalPosition.rawValue)
+    }
+    
+    var bearing:Double {
+        return self.point.bearingTo(Direction.None.point)
+    }
+    
+    static let None = Direction(horizontalPosition: .Middle, verticalPosition: .Middle)
+    static let Up = Direction(horizontalPosition: .Middle, verticalPosition: .Top)
+    static let Down = Direction(horizontalPosition: .Middle, verticalPosition: .Bottom)
+    static let Left = Direction(horizontalPosition: .Left, verticalPosition: .Middle)
+    static let Right = Direction(horizontalPosition: .Right, verticalPosition: .Middle)
+    
+    static let TopLeft = Direction(horizontalPosition: .Left, verticalPosition: .Top)
+    static let TopRight = Direction(horizontalPosition: .Right, verticalPosition: .Top)
+    static let BottomLeft = Direction(horizontalPosition: .Left, verticalPosition: .Bottom)
+    static let BottomRight = Direction(horizontalPosition: .Right, verticalPosition: .Bottom)
+}
+
+
+//MARK: Geometry
+
+extension CGPoint {
+    func distanceTo(point:CGPoint) -> CGFloat {
+        return sqrt(pow(point.x - self.x, 2) + pow(point.y - self.y, 2))
+    }
+    
+    func bearingTo(point:CGPoint) -> Double {
+        return atan2(Double(point.y - self.y), Double(point.x - self.x))
+    }
+    
+    func scalarProjectionWith(point:CGPoint) -> CGFloat {
+        return dotProductWith(point)/point.modulo
+    }
+    
+    func scalarProjectionPointWith(point:CGPoint) -> CGPoint {
+        let r = scalarProjectionWith(point) / point.modulo
+        return CGPoint(x: point.x * r, y: point.y * y)
+    }
+    
+    func dotProductWith(point:CGPoint) -> CGFloat {
+        return (self.x * point.x) + (self.y * point.y)
+    }
+    
+    var modulo:CGFloat {
+        return sqrt(self.x*self.x + self.y*self.y)
+    }
+    
+    func distanceToRect(rect: CGRect) -> CGFloat {
+        if rect.contains(self) {
+            return distanceTo(CGPoint(x: rect.midX, y: rect.midY))
+        }
+        let dx = max(rect.minX - self.x, self.x - rect.maxX, 0)
+        let dy = max(rect.minY - self.y, self.y - rect.maxY, 0)
+        if dx * dy == 0 {
+            return max(dx, dy)
+        } else {
+            return hypot(dx, dy)
+        }
+    }
+    
+    func normalizedPointForSize(screenSize:CGSize) -> CGPoint {
+        let x = 2 * (self.x / screenSize.width)
```

Comment:
Ok this function name was actually misleading, I added another function that performed the normalisation correctly. 
What this one does is normalize a distance based on the screensize. It is multiplied by two because the normalized coordinate system goes from (-1, 1)